src/components/LogTable.vue

```diff
@@ -107,19 +107,71 @@ export default {
               return;
             }
 
+
             switch (col.name) {
               case 'Log type':
                 where.type = { $iLike: `%${value}%` };
                 break;
               case 'App':
                 where.app = { name: { $iLike: `%${value}%` } };
                 break;
-              case 'User':
-                where.$or = [
-                  { user: { type: null, email: { $iLike: `%${value}%` } } },
-                  { data: { _userEmail: { $iLike: `%${value}%` } } }
-                ];
+
+              case 'User': {
+                const stringRegExp = '([-a-zA-Z0-9@:%_\\+.~#?&//=]*)';
+
+                const impersonateUserEmail = new RegExp(`^${stringRegExp} as ${stringRegExp}$`);
+                const impersonateUserStartAs = new RegExp(`^as ${stringRegExp}$`);
+                const impersonateUserEndAs = new RegExp(`^${stringRegExp} as$`);
+                const valueToLowerCase = String(value).toLowerCase();
+
+                if (impersonateUserEmail.test(valueToLowerCase)) {
```

Comment:
This whole logic doesn't feel DRY. Please look for opportunities to refactor it.

Reply:
@squallstar , I have refactored some parts of the code. can you please review it? Once you review it, I will ask Kathi to test the whole feature again.?

Reply:
@anerib you still haven't refactored to use a single `String.match`. This looks awful:

```js
const impersonateUserEmails = value.match(new RegExp(`^${regExToMatchEmailStr} as ${regExToMatchEmailStr}$`, 'i'));
const impersonateUserStartAsEmail = value.match(new RegExp(`^as ${regExToMatchEmailStr}$`, 'i'));
const impersonateUserEndAsEmail = value.match(new RegExp(`^${regExToMatchEmailStr} as$`, 'i'));
```

You should be using match once to grab both emails at once. Can you confirm you're aware of how RegExp capturing groups work?

See:

<img width="475" alt="image" src="https://user-images.githubusercontent.com/574210/176375803-44bb5ad1-127f-48b0-bfdb-21970f9e5808.png">

Reply:
I believe we both are not on the same page.  I am not using 3 regular expressions to fetch the 2 emails from "foo as bar". Here, added support for 3 kinds of strings filter as mentioned in the PR description. 

Let me add a comment on what each RegEx does:

1. `const impersonateUserEmails = value.match(new RegExp(`^${regExToMatchEmailStr} as ${regExToMatchEmailStr}$`, 'i'));`
// Here, `value` is a string only. Above will match the string "foo as bar" and store `foo as bar`, `foo`, and `bar` in `impersonateUserEmails`. `impersonateUserEmails` variable is being used in conditions. 


3. `const impersonateUserStartAsEmail = value.match(new RegExp(`^as ${regExToMatchEmailStr}$`, 'i'));`
// Here, `value` is a string only. Above will match the string which start with as E.g "as bar" and store `as bar`, and `bar` in `impersonateUserStartAsEmail`. `impersonateUserStartAsEmail` variable is being used in conditions. 

5. `const impersonateUserEndAsEmail = value.match(new RegExp(`^${regExToMatchEmailStr} as$`, 'i'));`
// Here, `value` is a string only. Above will match the string which end with as E.g "foo as" and store `foo as`, and `foo` in `impersonateUserEndAsEmail`. `impersonateUserEndAsEmail` variable is being used in conditions. 

This way admin can see filter the log with 3 formats which are 'email as email', 'email as' and 'as email'.

In addition to above, we wont be able to use `(/([^ ]+) as ([^ ]+)/)` as it will not extract emails correctly in case of multiple spaces in search. See below. 


![image](https://user-images.githubusercontent.com/101858104/176379469-752f4ef3-a5d6-4095-9aff-793c80cf9462.png)

Please, let me know if I am misunderstanding anything.

Reply:
@anerib thanks, I think that makes sense, in order to accept partial matches for each side of the expression your logic works and refactoring isn't required. I'll review your changes and see if it's good to go.